mayastor/src/bdev/dev/nvmx/uri.rs

```diff
@@ -286,6 +285,43 @@ impl CreateDestroy for NvmfDeviceTemplate {
 
     // nvme_bdev_ctrlr_create
     async fn destroy(self: Box<Self>) -> Result<(), Self::Error> {
+        // 1. Initiate controller shutdown, which shuts down all I/O resources
+        // of the controller.
+        let controller = NVME_CONTROLLERS
+            .lookup_by_name(self.get_name())
```

Comment:
Not sure it's an issue.
We access NVME_CONTROLLERS in similar manner in lots of our other async functions, all works fine.
Just a side note: lint mostly complains about ".await'ing with a mutex lock held", which is not the case here.

Reply:
It wouldn't be a lint issue, but what would happen if one async thread grabs the lock, then another async thread tries to grab the lock. At this point this is a deadlock because the other async fn will "block" the reactor and the initial thread won't run?

Reply:
If the first thread doesn't block forever while holding the lock, then it's not an issue, as the lock will be released upon leaving the lock scope and the second thread will proceed. We don't have any sleep/rescheduling with the lock held, we just grab the lock and do some 'atomic work' (I mean, not interrupted by other jobs).

Reply:
Yeah, I guess it's fine as long as there's no async code called while the lock is held.

Reply:
Yes.
And that's exactly what Lint blames about when it sees '.await' with a lock held.